File: 백준/Bronze/14563. 완전수/완전수.py
import sys
input=lambda:sys.stdin.readline().rstrip()
n=int(input())
arr=[*map(int,input().split())]
for i in range(n):
    t,pri_num=arr[i],0 if arr[i]==1 else 1
    for i in range(2,t):
        if t%i==0:
            pri_num+=i
    print("Perfect" if pri_num==t else "Deficient" if pri_num<t else "Abundant")

File: 백준/Bronze/14563. 완전수/완전수.cc
#include <iostream>
using namespace std;
int main(void) {
    int n, pri_num, elem;
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> elem;
        pri_num = elem == 1 ? 0 : 1;
        for (int j = 2; j < elem; j++) {
            if (elem % j == 0)
                pri_num += j;
        }
        cout << (pri_num == elem ? "Perfect" : (pri_num < elem ? "Deficient" : "Abundant")) << "\n";
    }
    return 0;
}